src/udev/net/ethtool-util.c

```diff
@@ -572,7 +593,15 @@ int ethtool_set_glinksettings(int *fd, const char *ifname, struct link_config *l
         if (link->port != _NET_DEV_PORT_INVALID)
               u->base.port = link->port;
 
-        u->base.autoneg = link->autonegotiation;
+        if (u->base.autoneg != link->autonegotiation)
+                u->base.autoneg = link->autonegotiation;
```

Comment:
Why do you add checking these value are not equivalent?

Reply:
avoid copying.